src/ShelfLayoutManager.Infrastructure/CabinetsDataService.cs

```diff
@@ -0,0 +1,116 @@
+using FluentResults;
+using Microsoft.EntityFrameworkCore;
+using ShelfLayoutManager.Core;
+using ShelfLayoutManager.Infrastructure.Converters;
+
+namespace ShelfLayoutManager.Infrastructure;
+
+/// <inheritdoc>
+public class CabinetsDataService(DbContext dbContext, ICabinetEntityConverter cabinetEntityConverter)
+    : ICabinetsDataService
+{
+    private readonly DbContext _dbContext = dbContext;
+
+    private readonly ICabinetEntityConverter _cabinetEntityConverter = cabinetEntityConverter;
+
+    public async Task<Result<Cabinet?>> GetAsync(long id, bool includeRows, bool includeLanes, CancellationToken ct) =>
+        _cabinetEntityConverter.ConvertToCabinet(
+            await GetReadQueryable(includeRows, includeLanes).Where(c => c.Id == id).FirstOrDefaultAsync(ct));
+
+    public async Task<Result<IList<Cabinet>>> ListAsync(bool includeRows, bool includeLanes, CancellationToken ct) =>
+        (await GetReadQueryable(includeRows, includeLanes).ToListAsync(ct))
+            .Select(_cabinetEntityConverter.ConvertToCabinet).OfType<Cabinet>().ToArray();
+
+    public async Task<Result> AssignSkuAsync(CabinetSkuAssignment cabinetSkuAssignment, CancellationToken ct)
+    {
+        // TODO(abalkar): Add validation for given assignment.
```

Comment:
This is also probably the most overloaded function in the codebase.
It has 3 responsibilities.
1. Set SKU assignment
2. Remove SKU assignment
3. Move SKU assignment

These could really be 3 separate functions and be less confusing to look at.
And easier to validate.